### DISCUSSION TITLE
Replacing Boost data structures with STL

### DISCUSSION THREAD

**Discussion by GYuvanShankar** [CGAL/cgal#6458] [General]
Many files still use boost::tuple, even though std::tuple provides similar features. Would it be worthwhile to change it?

**Comment by mglisse** [comment]
In places where it is an implementation detail, it would make sense to change it. Some implementations of std::tuple store objects in the reverse order of boost::tuple, so there is a slight risk that it impacts memory or performance, but it doesn't seem too likely.
In places where this is part of the public interface, changing it will break users' code. It may still make sense in the long run, but I don't know what the right strategy would be to minimize the trouble for users.

**Comment by GYuvanShankar** [comment]
Thanks. Ill try changing in a few instances.

**Comment by sloriot** [comment]
In the Kernel it won't be easy as `boost::tuple` does not need to know the complete type of objects stored while `std::tuple` does. I think I already did replace all the easy one.
